debezium-quarkus/extensions/outbox/runtime/src/main/java/io/debezium/outbox/quarkus/ExportedEvent.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.outbox.quarkus;
+
+import java.time.Instant;
+
+/**
+ * Describes an event that should be exported via the "outbox" table.
+ *
+ * @author Chris Cranford
+ */
+public interface ExportedEvent<I, P> {
```

Comment:
As discussed on the chat: the mapping of the `OutboxEvent` entity should be dynamically derived from the types for id and payload used in a given project. If those types are different across different event types, an error should be raised (down the road we might allow for several outbox tables of different configurations, should that need arise).

As one immediate follow-up, the derived mapping for the aggregate id an payload columns should be customizable by means of an option in application.properties. The same should be the case for the outbox table name.

Reply:
In the next commit, there are numerous configurable options.  In the deployment there is a class specific to build-time configuration and the runtime has its own as well.